src/components/PostGrid.jsx

```diff
@@ -1,9 +1,40 @@
-import React from "react";
-
+import React, {useState, useEffect} from "react";
+import { createClient } from '@supabase/supabase-js';
 import getAvatar from "../lib/getAvatar";
 
+
+// probably should move these to an env.
+const supabase = createClient('https://ibcwmlhcimymasokhgvn.supabase.co', 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJyb2xlIjoiYW5vbiIsImlhdCI6MTYyOTkzMDc3OCwiZXhwIjoxOTQ1NTA2Nzc4fQ.zcdbd7kDhk7iNSMo8SjsTaXi0wlLNNQcSZkzZ84NUDg')
+
+async function fetchVotesByRepo(repoName) {
+  const { data: recommendations, error } = await supabase
+    .from('recommendations')
+    .select('votes')
+    .eq('repo_name', repoName);
+
+    console.error(error);
+
+    return recommendations[0].votes ? recommendations[0].votes : 0;
+}
```

Comment:
Should create supbase.js file to house these crud operations and make them reuable.